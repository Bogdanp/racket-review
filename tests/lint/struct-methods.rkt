#lang racket/base

(require racket/generic
         racket/match)

(define-generics store
  {store-ref store})

(struct memory-store (ttl)
  #:methods gen:store
  [(define (store-ref s)
     (match-define (memory-store ttl) s)
     (void))])
