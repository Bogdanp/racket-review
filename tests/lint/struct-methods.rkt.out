/struct-methods.rkt:6:18:warning:identifier 'store?' is never used
/struct-methods.rkt:7:4:warning:identifier 'store-ref' is never used
/struct-methods.rkt:11:13:warning:identifier 'store-ref' is never used
/struct-methods.rkt:11:13:warning:identifier 'store-ref' shadows an earlier binding
/struct-methods.rkt:12:34:warning:identifier 'ttl' is never used
